test/snippet/argument_parser/custom_argument_parsing_enumeration.cpp

```diff
@@ -0,0 +1,46 @@
+#include <system_error>
+
+#include <seqan3/argument_parser/all.hpp>
+
+namespace seqan3::custom
+{
+// Specialise the seqan3::custom::argument_parsing data structure to enable parsing of std::errc.
+template <>
+struct argument_parsing<std::errc>
+{
+    // Specialise a mapping from an identifying string to the respective value of your type Foo.
+    static inline const std::unordered_map<std::string_view, std::errc> enumeration_names
+    {{
+            {"no_error", std::errc{}},
+            {"timed_out", std::errc::timed_out},
+            {"invalid_argument", std::errc::invalid_argument},
+            {"io_error", std::errc::io_error}
+    }};
+};
+
+} // namespace seqan3::custom
+
+int main(int argc, char const * argv[])
+{
+    std::errc value{};
+
+    seqan3::argument_parser parser{"my_program", argc, argv};
+
+    // Because of the argument_parsing struct and
+    // the static member function enumeration_names
+    // you can now add an option that takes a value of type std::errc:
+    parser.add_option(value, 'e', "errc", "Give me a std::errc value.", seqan3::option_spec::DEFAULT,
+                      seqan3::value_list_validator{(seqan3::enumeration_names<std::errc> | seqan3::views::get<1>)});
```

Comment:
Is the mechanism of using `seqan3::views::get` documented anywhere?

Reply:
What do you mean? This is a snippet that appears in the documentation of "how to specialise this". Is that documentation enough?

Reply:
I just think that most users will be surprised that you can iterate over values/keys of an associative container using `views::get<>`.
If this is not explained anywhere, I think it would be good to add a sentence as explanation. But this can also be done later if you add a card.